solarforecastarbiter/reports/figures/plotly_figures.py

```diff
@@ -0,0 +1,375 @@
+"""
+Functions to make all of the metrics figures for Solar Forecast Arbiter reports
+using Plotly.
+"""
+import calendar
+import datetime as dt
+from itertools import cycle
+import logging
+
+
+from bokeh import palettes
+import pandas as pd
+from plotly import __version__ as plotly_version
+import numpy as np
+
+
+from solarforecastarbiter import datamodel
+
+import plotly.graph_objects as go
+
+
+logger = logging.getLogger(__name__)
+PALETTE = (
+    palettes.d3['Category20'][20][::2] + palettes.d3['Category20'][20][1::2])
+
+
+PLOT_BGCOLOR = '#FFF'
+PLOT_MARGINS = {'l': 50, 'r': 50, 'b': 50, 't': 50, 'pad': 4}
+PLOT_LAYOUT_DEFAULTS = {
+    'autosize': True,
+    'height': 250,
+    'margin': PLOT_MARGINS,
+    'plot_bgcolor': PLOT_BGCOLOR,
+    'font': {'size': 14}
+}
+
+
+def configure_axes(fig, x_ticks, y_range):
+    """Applies plotly axes configuration to display zero line and grid.
+    Parameters
+    ----------
+    fig: plotly.graph_objects.Figure
+
+    x_range: list
+        List of values for the x axis, None will allow cause the plot to use
+        x values as labels and restrict to available data.
+    y_range: Tuple
+        Tuple of (min, max) to set the initial y range of the plot.
+    """
+    fig.update_xaxes(showline=True, linewidth=1, linecolor='black')
+    if x_ticks is not None:
+        fig.update_xaxes(ticks='outside', tickvals=x_ticks,
+                         range=(-.5, len(x_ticks)))
+    fig.update_yaxes(showgrid=True, gridwidth=1, gridcolor='#CCC')
+    fig.update_yaxes(showline=True, linewidth=1, linecolor='black')
+    fig.update_yaxes(ticks='outside', range=y_range)
+
+
+def construct_metrics_dataframe(metrics, rename=None):
+    """
+    Possibly bad assumptions:
+    * metrics contains keys: name, Total, etc.
+
+    Parameters
+    ----------
+    metrics : list of datamodel.MetricResults
+        Each metric dict is for a different forecast. Forecast name is
+        specified by the name key.
+    rename : function or None
+        Function of one argument that is applied to each forecast name.
+
+    Returns
+    -------
+    df: pandas.DataFrame
+        Dataframe of computed metrics for the report.
+    """
+
+    if rename:
+        f = rename
+    else:
+        def f(x): return x  # NOQA
+
+    data = []
+    for metric_result in metrics:
+        for mvalue in metric_result.values:
+            new = {
+                'name': metric_result.name,
+                'abbrev': f(metric_result.name),
```

Comment:
any idea why the abbreviate column is the same as name when I run the test report from the cli?

Reply:
I had neglected to pass the abbreviate function when generating the `metrics_json` template var in template.py.